frontend/src/reducer/notificationPage.js

```diff
@@ -49,42 +61,220 @@ export const serializeNotification = notification => {
   }
 }
 
+const hasSameAuthor = authorList => {
+  if (authorList.some(author => !author)) return false
+  return authorList.every((author, index) => {
+    if (index === 0) return true
+    return author.userId === authorList[index - 1].userId
+  })
+}
+
+const hasSameWorkspace = workspaceList => {
+  if (workspaceList.some(workspace => !workspace)) return false
+  return workspaceList.every((workspace, index) => {
+    if (index === 0) return true
+    return workspace.id === workspaceList[index - 1].id
+  })
+}
+
+const hasSameContent = notificationList => {
+  if (notificationList.some(notification => !notification.content)) return false
+  return notificationList.every((notification, index) => {
+    if (index === 0) return true
+    const previousContentId = notificationList[index - 1].type.includes(CONTENT_TYPE.COMMENT)
+      ? notificationList[index - 1].content.parentId
+      : notificationList[index - 1].content.id
+    const contentId = notification.type.includes(CONTENT_TYPE.COMMENT)
+      ? notification.content.parentId
+      : notification.content.id
+    return contentId === previousContentId
+  })
+}
```

Comment:
Same logic as previously

Reply:
This one is a lit bit more complex to use only some and I think this makes the code more difficult to read, so if you are ok I'll leave it like this.

Reply:
Ok then maybe this?

```suggestion
const hasSameContent = notificationList => {
  return notificationList.every((notification, index) => {
    if (index === 0) return true
    if (!notification.content) return false
    const previousContentId = notificationList[index - 1].type.includes(CONTENT_TYPE.COMMENT)
      ? notificationList[index - 1].content.parentId
      : notificationList[index - 1].content.id
    const contentId = notification.type.includes(CONTENT_TYPE.COMMENT)
      ? notification.content.parentId
      : notification.content.id
    return contentId === previousContentId
  })
}
```

Reply:
I tested and, sadly, it doesn't works

Reply:
Strange. Anyway don't waste too much time on this.